### ISSUE TITLE
Canvas + circleMarker popup invisible click area

### ISSUE THREAD

**Issue by jhzn** [Leaflet/Leaflet#5797]
## How to reproduce

- Leaflet version I'm using: **1.2.0**
- Browser (with version) I'm using: **Chrome Version 60.0.3112.113**
- OS/Platform (with version) I'm using: **Windows 10**

## What behaviour I'm expecting and which behaviour I'm seeing

The popup should only trigger when actually clicking on the circlemarker.
It works fine with markers.

The click area to prompt the popup is greater than the circlemarkers visible area.

## Minimal example reproducing the issue

https://jsfiddle.net/L6nqpz2c/9/

```javascript
//Using leaflet.js and leaflet.css @ v.1.2.0

var mymap = L.map('mapid', {
  renderer: L.canvas() //This must be used
}).setView([51.5, -0.09], 14);

L.tileLayer('https://api.tiles.mapbox.com/v4/{id}/{z}/{x}/{y}.png?access_token=pk.eyJ1IjoibWFwYm94IiwiYSI6ImNpejY4NXVycTA2emYycXBndHRqcmZ3N3gifQ.rJcFIG214AriISLbB6B5aw', {
  maxZoom: 14,
  attribution: 'Map data &copy; <a href="http://openstreetmap.org">OpenStreetMap</a> contributors, ' +
    '<a href="http://creativecommons.org/licenses/by-sa/2.0/">CC-BY-SA</a>, ' +
    'Imagery © <a href="http://mapbox.com">Mapbox</a>',
  id: 'mapbox.streets'
}).addTo(mymap);

var circleMarker1 = L.circleMarker([51.5, -0.092]).addTo(mymap);

//This circlemarker has an bigger clickable/invisible radius
var circleMarker2 = L.circleMarker([51.5, -0.09]).addTo(mymap);

var marker1 = L.marker([51.496, -0.092]).addTo(mymap);

//Works as expected with popup
var marker2 = L.marker([51.496, -0.09]).addTo(mymap);

circleMarker2.bindPopup("This popup triggers outside the circlemarker");

marker2.bindPopup("This behaves as expected");
```

**Comment by IvanSanchez** [comment]
This is a feature, not a bug. When the web browser is touch-capable, the hitboxes of all vector layers are enlarged by a hard-coded 10 pixels to allow users to use their unprecise fingers.

See https://github.com/Leaflet/Leaflet/blob/master/src/layer/vector/Path.js#L143

We currently have no way of setting different tolerances for mouse/finger/pen `PointerEvent`s, though.

**Comment by jhzn** [comment]
Ok. Interesting. Thanks for the answer. Why is there a difference in clicktolerance between rendering methods?

**Comment by CalvinWilliams1012** [comment]
@IvanSanchez I have been looking into this and it looks like leaflets very own @perliedman has made a plugin to do this with vectors available here: https://github.com/perliedman/leaflet-touch-helper

Maybe we should just add an option in `Path.js` for click tolerance amount with default 10 and use it in the `_clickTolerance` function instead of the static `10` that is currently there.

Either that or remove the `_clickTolerance` function and just add to the `PointerEvent`'s Width and Height, which is read only so it would need to be wrapped, which I believe `DomEvent.Pointer` is doing. If I am right I believe I would need to add an option then add that to the height/width when a `PointerEvent `is pushed onto `_pointers` which I believe happens in the `_globalPointer` functions in `DomEvent.Pointer`.

Sorry for the long posts but I wouldn't mind submitting a PR for either of these solutions _(Not completely sure if I am right on the PointerEvent solution, would like someone else to look over it.)_ just not sure which one is wanted.

**Comment by johnd0e** [comment]
> We currently have no way of setting different tolerances for mouse/finger/pen `PointerEvent`s, though.

It seems not so hard to fix, as we have this info in `_onClick` method, so we could pass it to `layer._containsPoint` here:

https://github.com/Leaflet/Leaflet/blob/61219d38ed1c41f933676e12a5070fc1818c5c03/src/layer/vector/Canvas.js#L343-L351

**Comment by hollowM1ke** [comment]
> ## What behaviour I'm expecting and which behaviour I'm seeing
> The popup should only trigger when actually clicking on the circlemarker. It works fine with markers.
> 
> The click area to prompt the popup is greater than the circlemarkers visible area.

This behaviour doesn't seem to be present anymore.

- Leaflet version: 1.9.2
- Browser version: 120.0.6099.225 (Official Build) (64-bit)
- OS: Windows 10

**Comment by Falke-Design** [comment]
@hollowM1ke thank you. It was already fixed in November 2017 with #5922
